internal/commands/app/create_inputs.go

```diff
@@ -153,55 +154,110 @@ func (i *createInputs) resolveLocalPath(ui terminal.UI, wd string) (string, erro
 }
 
 func (i *createInputs) resolveClusters(ui terminal.UI, client atlas.Client, groupID string) ([]dataSourceCluster, []string, error) {
-	clusters, err := client.Clusters(groupID)
-	if err != nil {
-		return nil, nil, err
-	}
+	if i.Template != "" {
+		clusters, err := client.Clusters(groupID)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		// If creating template app, only allow a 0 or 1 passed-in cluster names
+		switch len(i.Clusters) {
+		case 0, 1:
+		default:
+			return nil, nil, errors.New("template apps can only be created with one data source")
+		}
+
+		if len(clusters) == 0 {
```

Comment:
Do all template apps require a cluster?

Reply:
Yes they do. In the future, that may be a different story if there were to be a template that only had functions, for example.

but until that happens, it's just a lot easier to assume that template apps require a cluster